### DISCUSSION TITLE
Basic Auth with unsafe character in server url

### DISCUSSION THREAD

**Discussion by rucciva** [keplerproject/luarocks#1271] [Q&A]
Hi, how to pass basic auth username or password that has unsafe characters such as @ in `--server` ?
i tried to url-encode the password but luarocks seems to not decode it back when constructing basic auth base64 header.

**Comment by daurnimator** [comment]
The url you pass gets directly passed onto the downloader, a normal basic auth url would look like `https://user:password@example.com/`, to put in a character like `@` you would url encode it.
e.g. for a password `1@3` you might encode it as: `https://user:1%403@example.com/`
